### DISCUSSION TITLE
Collapse grouped application windows into one icon

### DISCUSSION THREAD

**Discussion by lawand-dot-io** [lawand-dot-io/taskbar#50] [Feature requests]
When grouping windows by application (which makes them all position next to each other), it could be useful to collapse this group of windows into a single icon for the application. Clicking this icon should somehow allow to select one of the windows to switch the focus to, perhaps similarly to the Windows taskbar where thumbnails and titles of those windows are displays to allow selecting one of them for focus.

This feature request has been split from https://github.com/lawand-dot-io/taskbar/discussions/11

Edit (merged from feature request https://github.com/lawand-dot-io/taskbar/discussions/57): Add the option that apps remain ordered the same way whether or not they are open, to develop muscle memory when opening apps since they always remain at the same position in Taskbar.

**Comment by lawand-dot-io** [comment]
Duplicate: https://github.com/lawand-dot-io/taskbar/discussions/136

**Comment by jnjmarks** [comment]
I think this is a very logical step and one that would certainly entice me to enter into a paid subscription for this app.
One of the key functionalities of the Windows taskbar is the grouping of open documents/windows of the same app into one app icon with the ability for the user to hover their mouse to select the nested document/window.
Please consider this for future devs! Thanks and apologies for the duplicate suggestion.

**Reply by lawand-dot-io** [reply]
Thanks for the input 👍

**Comment by camner** [comment]
As I keep many browser windows/tabs open, this is a must for me in order to make Taskbar useful.
<img width="2556" alt="SCR-20250315-sqed" src="https://github.com/user-attachments/assets/49202780-eebd-4fce-9d58-c831c8419b7f" />
This is what Taskbar looks like "out of the box"...it isn't very useful with so many browser windows appearing in Taskbar.

**Reply by lawand-dot-io** [reply]
I agree that it's difficult to work with so many windows, but even if I implemented this feature to collapse them into one icon, you'll still have the same issue once you hover that window and look for the correct browser window to switch to. In case of many windows, I think they should be divided between different spaces.

**Comment by lawand-dot-io** [comment]
If I implement this feature, I wonder what should happen when clicking on the icon of an app that has multiple windows. I understand that hovering would show all thumbnails and clicking one of them would focus a single window, but what about clicking the app icon itself.

Maybe bring forward all of the app's windows, or the most recent one. hmm

**Reply by fucosk** [reply]
Neither. 
What MS Windows does is that it shows the thumbnails of all the app's windows and then you have to click on the chosen preview window. It s 2-click action, but instead of the macOS's right-clicking on the icon and then selecting the title of the window, on MS Windows, you left-click on the icon and then left click on the thumbnail.
The left-click on the icon is actually redundant, because yous see the thumbnails also on hover. So you can navigate to the icon, wait a second and move the mouse over the chosen thumbnail.

**Reply by lawand-dot-io** [reply]
@fucosk Thanks for the input. This makes sense. I will keep it in mind. Yeah, I think I should always see how they do it on Windows whenever I am in doubt since they pretty much nailed the taskbar over there haha. Thanks. Cheers

**Reply by Faithfinder** [reply]
There is a slight difference on Windows between left click and hover - if you click, the previews stay on screen even if you hover away.

This (grouping, not clicking) would be the most important feature for me, much more important than start menu or hiding :)

**Comment by VolucellaBombylans** [comment]
A recent TaskBar downloader - this would be my top feature request, and definitely seal it for me when coming to the paid subscription this year. I currently use the Taskbar with app icons (no text/titles) - so it would be useful if there was a single icon per app (e.g., I currently have 5 notes open, and so my taskbar has 5 'notes' app icons in a row, which feels messy). 

Instead, if the above feature was included, every time I clicked (or hovered) on a single 'notes' app icon, ideally a list of open notes would appear, which would include their titles. 

I'd really appreciate this feature, as otherwise, TaskBar is by far the best taskbar app I've tested (including uBar and a couple of others) - this is the one thing still missing IMO!

**Reply by lawand-dot-io** [reply]
Thank you for your additional input, I will keep this in mind once I have time to implement this feature later on. Thanks

**Comment by matt1** [comment]
Please ensure that this feature can be turned off! :)

For me, the *whole reason* I don't like the default mac dock is that it hides what is open and tries to group things together so there is just one icon - I want to be able to see everything that is there and to be able to reach it in a single click so that multitasking between windows is fast and I am not having to hunt around looking for things hidden away in lists/submenus/icons etc.

**Reply by lawand-dot-io** [reply]
Of course, I will always try as hard as I can to never remove any functionality that existing users are using 👍

**Comment by kingepic84** [comment]
Oh this would be great to have and would absolutely push me to purchasing the full version, the pinned apps retaining position would be especially useful paired with the groups
